src/graphql/outlier.rs

```diff
@@ -586,3 +609,161 @@ fn latest_outlier_key(before: &str) -> Result<Vec<u8>> {
+            let to = earliest_key(&after)?;
+            iter_through_search_filter_nodes(iter, &to, cmp::Ordering::is_ge, filter, last)
+        } else {
+            iter_through_search_filter_nodes(iter, &[], always_true, filter, last)
+        }?;
+        Ok((nodes, has_more, false))
+    } else {
+        let first = first.unwrap_or(DEFAULT_CONNECTION_SIZE);
+        let iter = if let Some(after) = after {
+            let start = earliest_key(&after)?;
+            map.iter_from(&start, Direction::Forward)?
+        } else {
+            map.iter_forward()?
+        };
+
+        let (nodes, has_more) = if let Some(before) = before {
+            let to = latest_key(&before)?;
+            iter_through_search_filter_nodes(iter, &to, cmp::Ordering::is_le, filter, first)
+        } else {
+            iter_through_search_filter_nodes(iter, &[], always_true, filter, first)
+        }?;
+        Ok((nodes, false, has_more))
+    }
+}
+
+fn iter_through_search_filter_nodes<I>(
+    iter: I,
+    to: &[u8],
+    cond: fn(cmp::Ordering) -> bool,
+    filter: &Option<SearchFilterInput>,
+    len: usize,
+) -> Result<(Vec<(String, RankedOutlier)>, bool)>
+where
+    I: Iterator<Item = (Box<[u8]>, Box<[u8]>)>,
+{
+    let mut nodes = Vec::new();
+    let mut exceeded = false;
+    for (k, v) in iter {
+        if !(cond)(k.as_ref().cmp(to)) {
+            break;
+        }
+
+        let curser = BASE64.encode(&k);
+        let Some(node) = RankedOutlier::from_key_value(&k, &v)?.into() else {
+            continue;
+        };
+
+        if let Some(filter) = filter {
+            if let Some(time) = &filter.time {
+                let start = time.start;
+                let end = time.end;
+                if let Some(start) = start {
+                    if let Some(end) = end {
+                        if node.timestamp <= start.timestamp_nanos_opt().unwrap_or_default()
+                            || node.timestamp > end.timestamp_nanos_opt().unwrap_or_default()
+                        {
+                            continue;
+                        }
+                    } else if node.timestamp < start.timestamp_nanos_opt().unwrap_or_default() {
+                        continue;
+                    }
+                } else if let Some(end) = end {
+                    if node.timestamp > end.timestamp_nanos_opt().unwrap_or_default() {
+                        continue;
+                    }
+                }
+            }
+
+            if let Some(distance) = &filter.distance {
+                let start = distance.start;
+                let end = distance.end;
+                if let Some(start) = start {
+                    if let Some(end) = end {
+                        if node.distance < start || node.distance > end {
+                            continue;
+                        }
+                    } else if (((node.distance * 10.0).round() / 10.0) - start).abs() > 0.0 {
```

Comment:
Why is the comparison condition to the start filter value written this way?

Reply:
For the distance displayed in aide-web it seems like it's displayed in a precision of 1. However, the node.distance has a high precision. So I inferred that the user might search with number with precision 1 and changed the node.distance to a precision of 1. Also, since there were 0.0 distance values I decided not to use Epsilon in such cases since I would want to have the nodes exactly what the user searched. Would there be a better way to do this (for example: if start is what the user searched and 0.4 and node.distance was 0.44444....)?

Reply:
I understand what you mean, but the current code will 'continue' even if the distance is greater than the start(For example: distance:22.222 , start:10.0), so it needs further modification.

Reply:
Wouldn't the code move on to the next iteration? Since the difference would be greater than 0.0, I thought that it would skip this iteration and not push it into the vector ,then move onto the next iteration when the distance is greater than 0.0 since I don't want to break the loop yet to get all the nodes after I skip this one.

Reply:
The start and end of the filter are the range of distances you want to look up, so if you have a single condition of start, I think all distances greater than start should be targeted.